internal/compute/ximera.go

```diff
@@ -0,0 +1,125 @@
+package compute
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/celestiaorg/talis/internal/types"
+)
+
+// XimeraProvider implements the Provider interface for Ximera
+type XimeraProvider struct {
+	client *XimeraAPIClient
+}
+
+// NewXimeraProvider creates a new Ximera provider instance
+func NewXimeraProvider() (*XimeraProvider, error) {
+	cfg, err := InitXimeraConfig()
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize ximera config: %w", err)
+	}
+	client := NewXimeraAPIClient(cfg)
+	return &XimeraProvider{client: client}, nil
+}
+
+// ValidateCredentials validates the Ximera credentials
+func (p *XimeraProvider) ValidateCredentials() error {
+	// Try to list servers as a credential check
+	_, err := p.client.ListServers()
+	if err != nil {
+		return fmt.Errorf("ximera credential validation failed: %w", err)
+	}
+	return nil
+}
+
+// GetEnvironmentVars returns the environment variables needed for the provider
+func (p *XimeraProvider) GetEnvironmentVars() map[string]string {
+	return map[string]string{
+		"XIMERA_API_URL":             os.Getenv("XIMERA_API_URL"),
+		"XIMERA_API_TOKEN":           os.Getenv("XIMERA_API_TOKEN"),
+		"XIMERA_USER_ID":             os.Getenv("XIMERA_USER_ID"),
+		"XIMERA_HYPERVISOR_GROUP_ID": os.Getenv("XIMERA_HYPERVISOR_GROUP_ID"),
+		"XIMERA_PACKAGE_ID":          os.Getenv("XIMERA_PACKAGE_ID"),
```

Comment:
nit: the constants could be extracted.

Reply:
like defining them as constants on top of this file or some other appropriate place if it is used in multiple places